src/persist-client/src/schema.rs

```diff
@@ -76,4 +390,334 @@ mod tests {
         assert_eq!(SchemaId::try_from("h1".to_owned()), Ok(SchemaId(1)));
         assert!(SchemaId::try_from("nope".to_owned()).is_err());
     }
+
+    #[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
+    struct Strings(Vec<String>);
+
+    impl Codec for Strings {
+        type Schema = StringsSchema;
+        type Storage = ();
+
+        fn codec_name() -> String {
+            "Strings".into()
+        }
+
+        fn encode<B: BufMut>(&self, buf: &mut B) {
+            buf.put_slice(self.0.join(",").as_bytes());
+        }
+        fn decode<'a>(buf: &'a [u8], schema: &Self::Schema) -> Result<Self, String> {
+            let buf = std::str::from_utf8(buf).map_err(|err| err.to_string())?;
+            let mut ret = buf.split(",").map(|x| x.to_owned()).collect::<Vec<_>>();
+            // Fill in nulls or drop columns to match the requested schema.
```

Comment:
@ParkMyCar We might want to make the same drop columns fix in the Codec impl for Row/SourceData? Otherwise the scenario where a reader is still looking at an old schema wouldn't work